lldb/source/Plugins/Language/Swift/SwiftLanguage.cpp

```diff
@@ -1832,19 +1832,110 @@ SwiftLanguage::GetDemangledFunctionNameWithoutArguments(Mangled mangled) const {
   return mangled_name;
 }
 
-void SwiftLanguage::FilterForLineBreakpoints(
-    llvm::SmallVectorImpl<SymbolContext> &sc_list) const {
-  llvm::erase_if(sc_list, [](const SymbolContext &sc) {
-    // If we don't have a function, conservatively keep this sc.
-    if (!sc.function)
-      return false;
+namespace {
+using namespace swift::Demangle;
+struct AsyncInfo {
+  const Function *function;
+  NodePointer demangle_node;
+  std::optional<uint64_t> funclet_number;
+
+  /// Helper function for logging.
+  std::string to_str() const {
+    StreamString stream_str;
+    llvm::raw_ostream &str = stream_str.AsRawOstream();
+    str << "function = ";
+    if (function)
+      str << function->GetMangled().GetMangledName();
+    else
+      str << "nullptr";
+    str << ", demangle_node: " << demangle_node;
+    str << ", funclet_number = ";
+    if (funclet_number)
+      str << *funclet_number;
+    else
+      str << "nullopt";
+    return stream_str.GetString().str();
+  }
+};
 
-    // In async functions, ignore await resume ("Q") funclets, these only
-    // deallocate the async context and task_switch back to user code.
+/// Map each unique Function in sc_list to a Demangle::NodePointer, or null if
+/// demangling is not possible.
+llvm::SmallVector<AsyncInfo> GetAsyncInfo(llvm::ArrayRef<SymbolContext> sc_list,
+                                          swift::Demangle::Context &ctx) {
```

Comment:
Could this be a function that translates just one symbol context and then you use a std::algorithm to apply that on an entire vector?

Reply:
Sadly there is no `std::transform_if` or anything along those lines that help here.
We would need to apply a unique filter based on SymbolContext.function and then std::transform.